map_view/set-up-location-driven-geotriggers/src/main/java/com/esri/samples/set_up_location_driven_geotriggers/SetUpLocationDrivenGeotriggersController.java

```
@@ -0,0 +1,298 @@
+
+    // access the json of the walking route points
+    String polylineData = IOUtils.toString(getClass().getResourceAsStream(
+      "/set_up_location_driven_geotriggers/polyline_data.json"), StandardCharsets.UTF_8);
+    // create a polyline from the location points
+    Polyline locations = (Polyline) Geometry.fromJson(polylineData, SpatialReferences.getWgs84());
+
+    // create a new simulated location data source to replicate the path walked around the garden
+    simulatedLocationDataSource = new SimulatedLocationDataSource();
+    // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+    simulatedLocationDataSource.setLocations(
+      locations, new SimulationParameters(Calendar.getInstance(), 2.0, 0.0, 0.0));
+
+    simulatedLocationDataSource.startAsync();
+
+    // disable map view interaction, the location display will automatically center on the mock device location
+    mapView.setEnableMousePan(false);
+    mapView.setEnableKeyboardNavigation(false);
+
+  }
+
+  /**
+   * Gets information from the fence geotrigger notification information and the fence feature itself and adds the information
+   * to the UI either as a formatted string or as an image.
+   *
+   * @param fenceFeature                    the feature
+   * @param fenceGeotriggerNotificationInfo the fence geotrigger notification info
+   */
+  private void handleAddingFeatureInfoToUI(ArcGISFeature fenceFeature, FenceGeotriggerNotificationInfo fenceGeotriggerNotificationInfo) {
+
+    // fetch the fence feature's attachments
+    ListenableFuture<List<Attachment>> attachmentsFuture = fenceFeature.fetchAttachmentsAsync();
+    // listen for fetch attachments to complete
+    attachmentsFuture.addDoneListener(() -> {
+      // get the feature attachments
+      try {
+        List<Attachment> attachments = attachmentsFuture.get();
+        if (!attachments.isEmpty()) {
```

should we have an else if it is empty? A bit belts and braces, but this may fail silently without it